fdbserver/RestoreController.actor.cpp

```diff
@@ -73,9 +73,46 @@ ACTOR static Future<Void> checkRolesLiveness(Reference<RestoreControllerData> se
 void splitKeyRangeForAppliers(Reference<ControllerBatchData> batchData,
                               std::map<UID, RestoreApplierInterface> appliersInterf, int batchIndex);
 
+ACTOR Future<Void> sampleBackups(Reference<RestoreControllerData> self, RestoreControllerInterface ci) {
+	loop {
+		try {
+			RestoreSamplesRequest req = waitNext(ci.samples.getFuture());
+			TraceEvent(SevDebug, "FastRestoreControllerSampleBackups")
+			    .detail("SampleID", req.id)
+			    .detail("BatchIndex", req.batchIndex)
+			    .detail("Samples", req.samples.size());
+			if (req.batchIndex > self->batch.size()) {
+				TraceEvent(SevError, "FastRestoreControllerSampleBackupsInvalidBatchIndex")
+				    .detail("BatchIndex", req.batchIndex)
+				    .detail("InitializedBatches", self->batch.size());
+				continue;
```

Comment:
this one is legit because if it happens, SevError will be logged and correctness will fail. This actually should never happen.

The Line 92 should send a reply back because it is a duplicate message.